clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1132,6 +1136,7 @@ public void onSuccess(Void value) {
                 if (interceptors != null)
                     interceptors.onCommit(offsets);
                 completedOffsetCommits.add(new OffsetCommitCompletion(cb, offsets, null));
+                inFlightAsyncCommits.decrementAndGet();
```

Comment:
Can we wrap the `decrementAndGet` calls in a `try`/`finally` block in case something weird happens in the code that comes before it?

Reply:
Yes, who knows what `interceptors.onCommit` will do. Fixed in 1fa48f53f0e6f5a2a9821075fa053e01cba6b0b2.